src/sidebar/oauth-auth.js

```diff
@@ -79,14 +90,23 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
+    // The delay, in milliseconds, before we will poll again to see if it's
+    // time to refresh the access token.
+    var delay = 5000;
```

Comment:
5 seconds is a bit short IMO. Making it a bit longer (eg. 30s or so) will reduce the number of times tabs containing Hypothesis will wake up if in the background. I believe making the timeout longer will also allow the browser to increase the window of timeouts which are coalesced together.